src/lib.rs

```diff
@@ -51,16 +52,16 @@ impl SmolStr {
                 buf[$idx] = byte
             });
         }
-        s!(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21);
+        s!(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22);
         SmolStr(Repr::Inline {
-            len: len as u8,
+            len: unsafe { transmute(len as u8) },
```

Comment:
If `unsafe` should be avoided, a big match is also possible, and they generate identical code if opt level >= 1

Reply:
It might be nicer, but I'm not sure it's worth the hassle for `dev` builds. r? @Veykril

Reply:
I think I prefer the match (if we have the option to avoid unsafe we should probably go for it)

Reply:
Well I checked again with something like
```rust
fn from_u8(input: u8) -> Self {
        match input {
            u if u == Self::_V0 as u8 => Self::_V0,
            // ...
            _ => unreachable!(),
        }
    }
```
and it generates
```asm
cmp     r12b, 23
mov     eax, 23
cmovb   eax, r12d
```
while `transmute` would simply be
```asm
mov     eax, r12d
```
So this unsafe is (at this point) unavoidable